src/main/java/org/ecommerce/util/money/operations/UsdConverter.java

```diff
@@ -0,0 +1,39 @@
+package org.ecommerce.util.money.operations;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.ecommerce.util.JsonParser;
+import org.springframework.stereotype.Service;
+
+import java.math.BigDecimal;
+import java.util.Map;
+
+@Service
+public class UsdConverter {
+
+    private static ApiCurrencyConverterService apiCurrencyConverterService;
+
+    UsdConverter(ApiCurrencyConverterService apiCurrencyConverterService) {
+        UsdConverter.apiCurrencyConverterService = apiCurrencyConverterService;
+    }
+
+    /**
+     * @param from String Currency code of the source
+     * @param to String Currency code of the expected output
+     * @param amount BigDecimal amount that will be converted
+     * @return The converted equivalence of the amount from source currency to expected currency
+     * @throws JsonProcessingException If {@link #apiCurrencyConverterService} fails while parsing
+     * the response to {@link ApiCurrencyConverterResponse} inside of {@link JsonParser}
+     */
+
+    public static BigDecimal convertAmountFromTo(String from, String to, BigDecimal amount) throws JsonProcessingException {
```

Comment:
Missing the integration with order creation logic

Reply:
Here's the logic implemented https://github.com/algutierrez-griddynamics/e-commerce/commit/07f9404f0a60ccd720d38e7193067e3e6a868879